src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -148,15 +170,30 @@ def _generic_method_handler(
                     )
                 )
                 client_public_key = bytes_to_public_key(client_public_key_bytes)
-                shared_secret = generate_shared_key(
-                    self.server_private_key,
-                    client_public_key,
-                )
-                verify = verify_hmac(
-                    shared_secret, request.SerializeToString(True), hmac_value
-                )
-                if not verify:
-                    context.abort(grpc.StatusCode.UNAUTHENTICATED, "Access denied")
+                if self.server_private_key is not None:
```

Comment:
Can the `server_private_key` ever be `None` when the `AuthenticateServerInterceptor` is enabled?

Reply:
Shouldn't be but our state is optional bytes